cpp/ql/test/library-tests/qlcfg/cpp11.cpp

```diff
@@ -54,6 +54,9 @@ void skip_init() {
 void run_init() {
   int nonstatic;
   static int x1 = global_int;
+
+  // It makes no sense to initialize a static variable to the address of a
+  // non-static variable, but in principle it can be done:
```

Comment:
I'm sure I could come up with plausible code where this would be a reasonable thing to do.  I think the static would only hold that address for a very controlled duration, and the point of it being static would lie elsewhere in the program.